go-controller/pkg/ovn/kubevirt.go

```diff
@@ -110,18 +122,287 @@ func (oc *DefaultNetworkController) deleteDHCPOptions(pod *kapi.Pod) error {
+	if err != nil {
+		return fmt.Errorf("failed reading ovn annotation: %v", err)
+	}
+	switchNames, err := oc.getSwitchNames(pod)
+	if err != nil {
+		return fmt.Errorf("failed configuring pod routing when getting switch current and original name: %v", err)
+	}
+
+	// VM is running at the node that owns the subnet the point to point
+	// routing is not needed
+	if switchNames.Current == switchNames.Original {
+		// Delete not needed policies or static routes
+		if err := libovsdbops.DeleteLogicalRouterPoliciesWithPredicate(oc.nbClient, types.OVNClusterRouter, func(policy *nbdb.LogicalRouterPolicy) bool {
+			policyVMLabel, ok := policy.ExternalIDs[string(libovsdbops.VirtualMachineIndex)]
+			return ok && policyVMLabel == pod.Labels[kubevirtv1.VirtualMachineNameLabel]
+		}); err != nil {
+			return fmt.Errorf("failed configuring pod routing when deleting stale policies for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+		}
+		if err := libovsdbops.DeleteLogicalRouterStaticRoutesWithPredicate(oc.nbClient, types.OVNClusterRouter, func(route *nbdb.LogicalRouterStaticRoute) bool {
+			routeVMLabel, ok := route.ExternalIDs[string(libovsdbops.VirtualMachineIndex)]
+			return ok && routeVMLabel == pod.Labels[kubevirtv1.VirtualMachineNameLabel]
+		}); err != nil {
+			return fmt.Errorf("failed configuring pod routing when deleting stale static routes for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+		}
+		return nil
+	}
+	lrpName := types.GWRouterToJoinSwitchPrefix + types.GWRouterPrefix + pod.Spec.NodeName
+	lrpAddresses, err := util.GetLRPAddrs(oc.nbClient, lrpName)
+	if err != nil {
+		return fmt.Errorf("failed configuring pod routing when reading LRP %s addresses: %v", lrpName, err)
+	}
+
+	//TODO: Should we use a transaction so all the routing elements appear
+	//      at once at ovn db ?
+	for _, podIP := range podAnnotation.IPs {
+		podAddress := podIP.IP.String()
+		podCIDR := (&net.IPNet{IP: podIP.IP.Mask(podIP.Mask), Mask: podIP.Mask}).String()
+		// For interconnect a static route that matches the VM src ip is needed
+		// to ensure the egress policy is evaluated, the nexthop/output port is
+		// overriden by the policy but we need them to have one per VM
+		if config.OVNKubernetesFeature.EnableInterconnect {
+			originalNode, err := oc.watchFactory.GetNode(switchNames.Original)
+			if err != nil {
+				return err
+			}
+			transitSwitchPortAddrs, err := util.ParseNodeTransitSwitchPortAddrs(originalNode)
+			if err != nil {
+				return err
+			}
+			ipFamily := utilnet.IPFamilyOfCIDR(podIP)
+			transitSwitchPortAddr, err := util.MatchFirstIPNetFamily(ipFamily == utilnet.IPv6, transitSwitchPortAddrs)
+			if err != nil {
+				return err
+			}
+
+			enablePoliciesRoute := nbdb.LogicalRouterStaticRoute{
+				IPPrefix: podCIDR,
+				Nexthop:  transitSwitchPortAddr.IP.String(),
+				Policy:   &nbdb.LogicalRouterStaticRoutePolicySrcIP,
+				ExternalIDs: map[string]string{
+					kubevirt.OvnZoneExternalIDKey: kubevirt.OvnLocalZone,
+				},
```

Comment:
When is this route cleaned up?

Also using that external id only is a bit risky, it's an external id that we can potentially use for other things in the future.

Reply:
This is no longer the case, since we decide to use cluster wide static routes, and I think they are not goin to be removed since we have just one per zone and do not affect the other traffic.